test-resources/ping.clj

```diff
@@ -0,0 +1,39 @@
+(ns ping
+  "This provides an example of a middleware, including a deferred handler
+  that's not loaded until called. This is representative of how cider-nrepl
+  and refactor-nrepl handles deferred loading"
```

Comment:
Btw, I think `refactor-nrepl` doesn't really defer anything, although this might have changed. It actually provides all its ops in a single middleware, which is somewhat different from what nREPL and `cider-nrepl` do. End of random comment. :-)

Reply:
I found out the hard way what does/doesn't do deferred loading :)

This handles deferred loading for `clean-ns`, for example.

https://github.com/clojure-emacs/refactor-nrepl/blob/dbafd6e686402dc020d7026e21ee556812fc51da/src/refactor_nrepl/middleware.clj#L109-L111

not all the ops do this, but a bunch of them do.

Reply:
Got it. I meant they don't use the same deferring that cider-nrepl does. Before we added the current macros, cider-nrepl was also just using `delay` for some requires.

Reply:
Right.